ui/app/components/auth/page.ts

```diff
@@ -118,4 +268,18 @@ export default class AuthPage extends Component<Args> {
     this.mfaAuthData = null;
     this.mfaErrors = '';
   }
+
+  // HELPERS
+  private filterVisibleMountsByType(authTypes: string[]) {
+    const tabs: UnauthMountsByType = {};
+    for (const type of authTypes) {
+      // adds visible mounts for each type, if they exist
+      tabs[type] = this.visibleMountsByType?.[type] || null;
+    }
+    return tabs;
+  }
+
+  private constructViews(view: FormView, tabData: UnauthMountsByType | null) {
+    return { view, tabData };
```

Comment:
`view` is actually never used anywhere and only included it to make the form view logic easier to navigate. Without it, the `dropdown` view has to be deduced whether or not `tabData` exists. So for state that has an alternate view, but no tabs it would be `alternateView = { tabData: null }`. It felt strange to rely on an object of that shape for a truthy value. Alternatively, I could add a key like `hasAlternateView` but at that point it just seemed like I might as well say what said alternate view is to ease navigating these chaotic logic combinations 😅